### PR TITLE
adding more unit test

### PR DESCRIPTION
**Pull request by robertsirc**
Adding more unit test, covering mostly the negative. 

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [X] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/kube/ready_test.go [hunk 1]
```diff
@@ -20,16 +20,212 @@ import (
...
+			},
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &ReadyChecker{
+				client:        tt.fields.client,
+				log:           tt.fields.log,
```
Comment (mattfarina) [inline pkg/kube/ready_test.go:213]: If `log` is always the same, you don't need a variable for it. You can use set it here.

File: pkg/kube/ready_test.go [hunk 2]
```diff
@@ -521,6 +717,144 @@ func Test_ReadyChecker_serviceReady(t *testing.T) {
 	}
 }
 
+func Test_ReadyChecker_crdBetaReady(t *testing.T) {
```
Comment (mattfarina) [inline pkg/kube/ready_test.go:720]: I see there are different test functions for different types against the same `ReadyChecker`. The different functions are testing different internal functions. Should these all be run through `IsReady` (the public function that calls these internal ones)? Thoughts?

Comment (robertsirc) [reply pkg/kube/ready_test.go:720]: You are correct, I don't like how this is written currently but I wanted to cover with test first then start refactoring. As I don't want to break anything. IsReady is huge and needs to be paired down.

File: pkg/kube/ready_test.go (hunk 1 above)
Comment (robertsirc) [reply pkg/kube/ready_test.go:213]: This is a stub out I am sure I will refactor this out later.

**Review by mattfarina — APPROVED** [review]
